src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -144,7 +144,19 @@ public class DevCanvasResultSourceService : IResultSourceService
         /// <inheritdoc/>
         public System.Threading.Tasks.Task InitializeAsync()
         {
-            // TODO: Remove this when merging into main.
+            return IsActiveAsync();
+        }
+
+        /// <inheritdoc/>
+        public Task<Result> IsActiveAsync()
+        {
+            // TODO: remove below to release to general users.
+            string devcanvasKey = (string)Registry.GetValue("HKEY_CURRENT_USER\\Software\\Microsoft\\VisualStudio\\devcanvas", "sampleKey", null);
+            if (string.IsNullOrEmpty(devcanvasKey))
+            {
+                return System.Threading.Tasks.Task.FromResult(Result.Failure("Not a DevCanvas user."));
```

Comment:
`Result.Failure()` does not seem correct here - we have successfully not activated for the user.  This is by design - I would not categorize by-design behavior as a failure.

Reply:
In a way the other result services do the same pattern, if the service is not active it returns a `Result.Failure` [example](https://github.com/microsoft/sarif-visualstudio-extension/blob/4818b476364eececfaf4dbe0cd3bb2e7eede73f2/src/Sarif.Viewer.VisualStudio.ResultSources.GitHubAdvancedSecurity.Core/Services/GitHubSourceService.cs#L189)

Also when the result source services are being initialized, if the result is not a failure it is added to the list of services that then get hooked into a bunch of events. [link](https://github.com/microsoft/sarif-visualstudio-extension/blob/4818b476364eececfaf4dbe0cd3bb2e7eede73f2/src/Sarif.Viewer.VisualStudio.ResultSources.Factory.Core/ResultSourceFactory.cs#L98C13-L98C13). Thought that it might be weird to constantly have a result source service be hooked into events just to swallow everything.